server/pulp/server/managers/consumer/agent.py

```diff
@@ -30,6 +33,12 @@
 from pulp.server.managers import factory as managers
 from pulp.server.async.task_status_manager import TaskStatusManager
 from pulp.server.agent import Context
+from pulp.server.async.tasks import Task
+
+
+QUEUE_DELETE_DELAY = 600  # 10 min.
```

Comment:
I think this is fine for now. The issue was first noticed by QE so in terms of tests it may be too long. This is already a huge improvement so lets leave it alone for now.